src/pages/billing/NewBillingProjectWizard/AzureBillingProjectWizard/AzureSubscriptionStep.test.ts

```diff
@@ -0,0 +1,176 @@
+      onManagedAppSelected: onManagedAppSelectedEvent,
+      inputDebounce: 0,
+    }
+    renderResult = render(h(AzureSubscriptionStep, {
+      ...defaultProps,
+      onSubscriptionIdChanged: newId => {
+        renderResult.rerender(h(AzureSubscriptionStep, { ...defaultProps, subscriptionId: newId }))
+      },
+      ...props
+    }))
+  }
+
+  beforeEach(() => {
+    jest.resetAllMocks()
+    // Don't show expected error responses in logs
+    jest.spyOn(console, 'error').mockImplementation(() => {})
+
+    // Arrange
+    onManagedAppSelectedEvent = jest.fn()
+  })
+
+  const captureEvent = jest.fn()
+  const invalidUuidError = 'Subscription id must be a UUID'
+  const noManagedApps = 'Go to the Azure Marketplace' // Can only test for complete text in an element, in this case the link.
+
+  it('has the correct initial state', () => {
+    renderAzureSubscriptionStep({})
+    // Assert
+    verifyDisabled(getManagedAppInput())
+  })
+
+  it('validates the subscription ID', async () => {
+    // Arrange
+    renderAzureSubscriptionStep({})
+    // Mock managed app Ajax call, should not be called
+    const listAzureManagedApplications = jest.fn(() => Promise.resolve())
+    asMockedFn(Ajax).mockImplementation(() => ({
+      Billing: { listAzureManagedApplications } as Partial<AjaxContract['Billing']>,
+      Metrics: { captureEvent } as Partial<AjaxContract['Metrics']>
+    } as Partial<AjaxContract> as AjaxContract))
+
+    // Assert - UUID error message should not initially be visible, even though subscription ID field is empty.
+    expect(screen.queryByText(invalidUuidError)).toBeNull()
+
+    // Act - Supply invalid UUID
+    fireEvent.change(getSubscriptionInput(), { target: { value: 'invalid UUID' } })
+
+    // Assert
+    await waitFor(() => expect(screen.queryByText(invalidUuidError)).not.toBeNull())
+    verifyDisabled(getManagedAppInput())
+    expect(listAzureManagedApplications).not.toHaveBeenCalled()
+    expect(captureEvent).not.toHaveBeenCalled()
+  })
+
+  const noManagedAppsTestCase = async listAzureManagedApplications => {
+    const subscriptionId = uuid()
+    asMockedFn(Ajax).mockImplementation(() => ({
+      Billing: { listAzureManagedApplications } as Partial<AjaxContract['Billing']>,
+      Metrics: { captureEvent } as Partial<AjaxContract['Metrics']>
+    } as Partial<AjaxContract> as AjaxContract))
+
+    // Act - Supply valid UUID
+    fireEvent.change(getSubscriptionInput(), { target: { value: subscriptionId } })
+
+    // Assert
+    await waitFor(() => expect(listAzureManagedApplications).toHaveBeenCalledWith(subscriptionId, false))
+    await screen.findByText(noManagedApps)
+    expect(screen.queryByText(invalidUuidError)).toBeNull()
+    verifyDisabled(getManagedAppInput())
+    expect(onManagedAppSelectedEvent).not.toHaveBeenCalled()
+  }
+
+  it('shows the spinner overlay while the call to list managed apps is in progress', async () => {
+    renderAzureSubscriptionStep({})
```

Comment:
The noManagedAppsTestCase function was testing that the setIsBusy function in the props was called.
Since the spinner was fixed, we no longer need that function, so I had to add a separate test case in order to check the page was showing the spinner while the call was in progress.